Documentation/Profiling/IL Rewriting Basics.md

```diff
@@ -0,0 +1,23 @@
+# IL Rewriting Basics
+
+## Intro
+One of the common use cases of the `ICorProfiler*` interfaces is to perform IL rewriting. Some possible reasons a profiler would want to rewrite IL:
+- Inspecting interesting process state
+  - Capturing exception state
+  - Inspecting managed objects
+  - Inspecting function arguments/return values
+- Injecting method hooks at the start/end of the method that call in to another managed library
```

Comment:
Could be used for injecting method at specific point, for instance for "code coverage instrumentation"?

Reply:
> ELT hooks 

From guide seem that ELT is too coarse(enter/exit func) for collect line/branch...but thank's for infos!
https://github.com/dotnet/coreclr/blob/master/Documentation/Profiling/davbr-blog-archive/ELT%20Hooks%20-%20The%20Basics.md